ext/src/ruby_api/wasi_ctx_builder.rs

```diff
@@ -222,20 +225,32 @@ impl WasiCtxBuilder {
         rb_self
     }
 
+    /// @yard
+    /// Make the context deterministic. See See https://github.com/Shopify/deterministic-wasi-ctx for more details
+    /// @def add_determinism()
+    /// @return [WasiCtxBuilder] +self+
+    pub fn add_determinism(rb_self: RbSelf) -> RbSelf {
+        let mut inner = rb_self.inner.borrow_mut();
+        inner.deterministic = true;
+        rb_self
+    }
+
     pub fn build(ruby: &Ruby, rb_self: RbSelf) -> Result<WasiCtx, Error> {
-        let mut builder = wasi_common::sync::WasiCtxBuilder::new();
+        let mut builder = wasmtime_wasi::WasiCtxBuilder::new();
         let inner = rb_self.inner.borrow();
 
         if let Some(stdin) = inner.stdin.as_ref() {
             match stdin {
                 ReadStream::Inherit => builder.inherit_stdin(),
-                ReadStream::Path(path) => {
-                    builder.stdin(file_r(ruby.get_inner(*path)).map(wasi_file)?)
-                }
+                ReadStream::Path(path) => builder.stdin(
+                    file_r(ruby.get_inner(*path))
+                        .map(|path| FileStdinStream::new(FileInputStream::new(&path, 0)))?,
```

Comment:
I cannot figure out how to construct `FileInputStream`. The link for the `&File` type in https://docs.rs/wasmtime-wasi/31.0.0/wasmtime_wasi/struct.FileInputStream.html#method.new doesn't exist. Looking at the underlying code, the `File` type used does not appear to be exported by `wasmtime-wasi`. We may want to write our own equivalent of `FileInputStream`.